### PR TITLE
[HOLD] Added Help Client to bootstrapper

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
- jQuery no longer a dependency for BBHelp
- Adjusted the `skyuxconfig.json` file to include a "packages" attribute, to store all three module's configuration. (It seemed easier to implement, and will flex well in the future if we need to add more modules.)
```
"packages": {
  "auth": true,
  "help": {},
  "omnibar": {}
}
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "@angular/platform-browser-dynamic": "2.4.2",
     "@angular/router": "3.4.2",
     "@blackbaud/auth-client": "blackbaud/auth-client",
+    "@blackbaud/help-client": "blackbaud/help-client",
```
Comment (Blackbaud-SteveBrush) [inline package.json]: I'm unsure if this is the correct way to implement a namespaced module...

Comment (Blackbaud-BobbyEarl) [inline package.json]: This is the correct way until we being publishing to npm.

File: runtime/bootstrapper.ts [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
+      }
+
+      // BBHelp init
+      if (SkyAppBootstrapper.bootstrapConfig.help) {
+        promises.push(BBHelp.load(SkyAppBootstrapper.bootstrapConfig.help));
       }
     }
 
-    return Promise.resolve();
+    return Promise.all(promises);
```
Comment (Blackbaud-BobbyEarl) [inline runtime/bootstrapper.ts]: Is the processBootstrapConfig resolved if auth, omnibar, and help are all false/not configured?

File: lib/sky-pages-bootstrap-generator.js [hunk 3]
```diff
@@ -4,16 +4,11 @@
...
-SkyAppBootstrapper.bootstrapConfig = {
-  omnibar: ${JSON.stringify(skyPagesConfig.omnibar)},
-  auth: ${JSON.stringify(skyPagesConfig.auth)}
-};
-
-`;
+  if (skyPagesConfig.packages) {
+    bootstrap = `
+      import { SkyAppBootstrapper } from '${skyPagesConfig.runtimeAlias}';
+      SkyAppBootstrapper.bootstrapConfig = ${JSON.stringify(skyPagesConfig.packages)};
```
Comment (Blackbaud-BobbyEarl) [inline lib/sky-pages-bootstrap-generator.js]: I can appreciate the efficiency by not having to specify auth, help, and omnibar in two places (here and `runtime/bootstrap.ts`, which has to stay).  

Doing this means we would have to instruct our current consumers to update their `skyuxconfig.json`, which I'm not sure is worth the efficiencies gained.  Generally speaking, we've found being very explicit in what options / configuration we allow has worked best.

Comment (Blackbaud-SteveBrush) [inline lib/sky-pages-bootstrap-generator.js]: Fair enough. I'll revert to a more deliberate approach (your tests also would have to be rewritten to accommodate, so probably not worth the effort).

**Pull request closed without merging**